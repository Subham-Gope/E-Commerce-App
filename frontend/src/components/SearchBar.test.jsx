// import { describe, test, expect, vi } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import { MemoryRouter } from "react-router-dom";
// import ShopContext from "../context/ShopContext";
// import SearchBar from "./SearchBar";

// // mocking the assets import
// vi.mock("../assets/assets", () => ({
//   assets: {
//     search_icon: "search_icon.png",
//     cross_icon: "cross_icon.png",
//   },
// }));

// // mock context
// const defaultMockContext = {
//   search: "",
//   setSearch: vi.fn(),
//   showSearch: true,
//   setShowSearch: vi.fn(),
// };

// // render with context function
// const renderWithContext = (value, route, ui) =>
//   render(
//     <MemoryRouter initialEntries={[route]}>
//       <ShopContext.Provider value={value}>{ui}</ShopContext.Provider>
//     </MemoryRouter>
//   );

// describe("Testing the SearchBar.jsx file", () => {
//   test("renders the searchBar", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     // rendering and setting the snapshot matching
//     const { container } = renderWithContext(
//       defaultMockContext,
//       "/collection/shirt",
//       <SearchBar />
//     );
//     expect(container).toMatchSnapshot();

//     // checking the input , texts, img
//     const input = screen.getByPlaceholderText(/search/i);
//     expect(input).toBeInTheDocument();
//     const crossIcon = screen.getByAltText(/cross icon/i);
//     expect(crossIcon).toBeInTheDocument();
//     const searchIcon = screen.getByAltText(/cross icon/i);
//     expect(searchIcon).toBeInTheDocument();

//     // Initiating the user interaction
//     await user.click(crossIcon);
//     expect(defaultMockContext.setShowSearch).toHaveBeenCalled();
//   });
// });
