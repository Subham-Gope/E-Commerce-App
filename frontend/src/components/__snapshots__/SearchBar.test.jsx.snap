// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the SearchBar.jsx file > renders the searchBar 1`] = `
<div>
  <div
    class="border-t border-b bg-gray-50 text-center"
  >
    <div
      class="inline-flex items-center justify-center border border-gray-400 px-5 py-2 my-5 mx-3 rounded-full w-3/4 sm:w-1/2"
    >
      <input
        class="flex-1 outline-none bg-inherit text-sm"
        placeholder="Search"
        type="text"
        value=""
      />
      <img
        alt="search icon"
        class="w-4"
        src="search_icon.png"
      />
    </div>
    <img
      alt="cross icon"
      class="inline w-3 cursor-pointer"
      src="cross_icon.png"
    />
  </div>
</div>
`;
